hphp/runtime/vm/jit/irlower-cmp.cpp

```diff
@@ -241,16 +241,23 @@ void cgCmpDbl(IRLS& env, const IRInstruction* inst) {
 
   auto& v = vmain(env);
   auto const sf = v.makeReg();
-  auto const tmp1 = v.makeReg();
-  auto const tmp2 = v.makeReg();
 
   assert(inst->src(0)->type() <= TDbl);
   assert(inst->src(1)->type() <= TDbl);
 
   v << ucomisd{s0, s1, sf};
-  v << cmovq{CC_A, sf, v.cns(-1), v.cns(1), tmp1};
-  v << cmovq{CC_NE, sf, v.cns(0), tmp1, tmp2};
-  v << cmovq{CC_P, sf, tmp2, v.cns(-1), d};
+  if (arch() == Arch::ARM) {
+    auto const tmp = v.makeReg();
+    // On ARM, CC_L => lt, which for FP is true for lower or unordered
+    v << cmovq{CC_L, sf, v.cns(1), v.cns(-1), tmp};
+    v << cmovq{CC_E, sf, tmp, v.cns(0), d};
+  } else {
+    auto const tmp1 = v.makeReg();
+    auto const tmp2 = v.makeReg();
+    v << cmovq{CC_A, sf, v.cns(-1), v.cns(1), tmp1};
+    v << cmovq{CC_NE, sf, v.cns(0), tmp1, tmp2};
+    v << cmovq{CC_P, sf, tmp2, v.cns(-1), d};
+  }
```

Comment:
I can actually implement this for ARM using the same three cmovq with the exception of the third which will use a CC_O for ARM vs. the CC_P. However, implementing it with two cmovq's leverages ARM's special unordered condition codes to eliminate the third explicit check for an unordered result. I assumed it's OK to have some lightly platform-specific code in here base another precedent earlier in the file. See 214-226 above.

With that said, I can easily revert back to using the three cmovqs for ARM too and then I'll just implement some special CC codes with switched implementations for the different platforms in their abi-xx files. Just let me know.